mlrun/projects/pipelines.py

```diff
@@ -547,7 +533,7 @@ def save(cls, project, workflow_spec: WorkflowSpec, target, artifact_path=None):
     @classmethod
     def run(
         cls,
-        project,
+        project,  # type: mlrun.projects.MlrunProject
```

Comment:
i think you can use type annotation with a string

Reply:
You should:

```
from .projects import MLRunProject
```

And then use it as type hint.

Reply:
This will cause circular imports, using `project: "mlrun.projects.MlrunProject"`.